opentracing-cdi-example/src/main/java/io/opentracing/contrib/cdi/example/SpanContextProducer.java

```diff
@@ -0,0 +1,18 @@
+package io.opentracing.contrib.cdi.example;
+
+import io.opentracing.SpanContext;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+public class SpanContextProducer {
+    @Inject
+    HttpServletRequest request;
+
+    @Produces
+    public SpanContext produceSpanContext() {
+        return (SpanContext) request.getAttribute(TracingFilter.SERVER_SPAN_CONTEXT);
```

Comment:
Another point to discuss: should the integration provide a producer for `ActiveSpan`, for no-async applications? If so, what should it return?

```java
    @Produces
    public ActiveSpan produceActiveSpan(InjectionPoint ip) {
        return GlobalTracer.get().activeSpan();
    }
```
(this would require whoever started the span to deactivate it)

Or:
```java
    @Produces
    public ActiveSpan produceActiveSpan(InjectionPoint ip) {
        ActiveSpan parent = GlobalTracer.get().activeSpan();
        return GlobalTracer
                .get()
                .buildSpan(ip.getMember().getName())
                .asChildOf(parent.context())
                .startActive();
    }
```
(this last would *require* the place where it gets injected to `deactivate` the span)